File: benchmarks/sieve/sieve.lua
local function sieve(n)
    local isPrime = {}
    for i = 0, n do
        isPrime[i] = true
    end
    isPrime[0] = false
    isPrime[1] = false

    local i = 2
    while i * i <= n do
        if isPrime[i] then
            for j = i * i, n, i do
                isPrime[j] = false
            end
        end
        i = i + 1
    end

    local count = 0
    for i = 2, n do
        if isPrime[i] then
            count = count + 1
        end
    end
    return count
end

local n = 6000000
local count = sieve(n)
print(string.format("primes up to %d: %d", n, count))

File: benchmarks/sieve/sieve.js
function sieve(n) {
    const isPrime = new Array(n + 1).fill(true);
    isPrime[0] = false;
    isPrime[1] = false;

    for (let i = 2; i * i <= n; i++) {
        if (isPrime[i]) {
            for (let j = i * i; j <= n; j += i) {
                isPrime[j] = false;
            }
        }
    }

    let count = 0;
    for (let i = 2; i <= n; i++) {
        if (isPrime[i]) {
            count++;
        }
    }
    return count;
}

const n = 6000000;
const count = sieve(n);
console.log(`primes up to ${n}: ${count}`);
